rusk-wallet/src/dat.rs

```diff
@@ -42,7 +42,34 @@ impl DatFileVersion {
     /// Checks if the file version is older than the latest Rust Binary file
     /// format
     pub fn is_old(&self) -> bool {
-        matches!(self, Self::Legacy | Self::OldWalletCli(_))
+        match self {
+            Self::Legacy | Self::OldWalletCli(_) => true,
+            Self::RuskBinaryFileFormat(version) => version < &LATEST_VERSION,
+        }
+    }
+}
+
+pub(crate) fn read_salt(
+    version: DatFileVersion,
+    bytes: &[u8],
+) -> Result<Option<[u8; 32]>, Error> {
+    match version {
+        DatFileVersion::Legacy | DatFileVersion::OldWalletCli(_) => Ok(None),
+        DatFileVersion::RuskBinaryFileFormat(version)
+            if version_without_pre_higher(version) < (0, 0, 2, 0) =>
+        {
+            Ok(None)
+        }
+        DatFileVersion::RuskBinaryFileFormat(_) => {
+            if let Some(salt_bytes) = bytes.get(12..12 + 32) {
```

Comment:
```suggestion
            if let Some(salt_bytes) = bytes.get(12..12 + SALT_SIZE) {
```

Let's define the salt-size, the IV-size, and the PBKDF2 rounds as constants to highlight them as specifications. Then we can also reuse the const values instead of the raw values in multiple places